### PR TITLE
Add analytics tracking for room creation

### PR DESCRIPTION
**Pull request by Johennes**
This adds screen and event tracking for room and DM creation.

### CODE REVIEW

**Comment by Johennes** [comment]
CC @giomfo because I think you might have had this on your list, too.

File: ElementX/Sources/Services/Analytics/Analytics.swift [hunk 1]
```diff
@@ -125,4 +116,11 @@ extension Analytics {
         let event = AnalyticsEvent.MobileScreen(durationMs: milliseconds, screenName: screen.screenName)
         client.screen(event)
     }
+
+    /// Capture an event in the `client`.
+    /// - Parameter event: The event to capture.
+    func capture(event: AnalyticsEventProtocol) {
```
Comment (pixlwave) [inline ElementX/Sources/Services/Analytics/Analytics.swift:122]: Please can we keep this private, instead adding public methods for tracking events like we do on EI:
[trackCreatedRoom](https://github.com/vector-im/element-ios/blob/2831265badf17c77f64e228b5372beb150d8050a/Riot/Modules/Analytics/Analytics.swift#L363)

This has the advantage of not needing to import AnalyticsEvents everywhere we're going to track things so it should be easier for forks to customise analytics with different schemas.

**Review by pixlwave — CHANGES_REQUESTED** [review]
Nice, its good to see PostHog starting to be used!

File: ElementX/Sources/Services/Analytics/Analytics.swift (hunk 1 above)
Comment (Johennes) [reply ElementX/Sources/Services/Analytics/Analytics.swift]: Ah ok. Was already wondering what the additional abstraction layer in the old app was good for. Makes sense and have changed.

**Review by pixlwave — APPROVED** [review]
LGTM 😎

File: ElementX/Sources/Services/Analytics/Analytics.swift (hunk 1 above)
Comment (pixlwave) [reply ElementX/Sources/Services/Analytics/Analytics.swift]: Super thanks

**Pull request merged**